src/main/java/core/basesyntax/service/strategy/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.service.counter.OperationHandler;
+import core.basesyntax.service.transaction.FruitTransaction;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+
+    private Map<String, OperationHandler> handlersMap;
+
+    public OperationStrategyImpl(Map<String, OperationHandler> handlersMap) {
+        this.handlersMap = handlersMap;
+    }
+
+    @Override
+    public OperationHandler getOperationType(FruitTransaction fruitTransaction) {
+        return handlersMap.get(fruitTransaction.getOperation().getCode());
```

Comment:
```suggestion
        return handlersMap.get(fruitTransaction.getOperation());
```
Better have Operation as Key in the handlersMap